apiclient/exceptions.py

```diff
@@ -18,10 +20,14 @@ class ResponseParseError(APIClientError):
 class APIRequestError(APIClientError):
     """Exception to denote that something went wrong when making the request."""
 
-    def __init__(self, message, status_code: OptionalInt = None, info: str = ""):
-        self.message = message
-        self.status_code = status_code
-        self.info = info
+    message: str = ""
+    status_code: OptionalInt = None
+    info: str = ""
```

Comment:
you can remove these definitions.  They aren't used.

Reply:
👌

Reply:
This is needed for child classes. For example, to set values ​​that will override what is passed.
For example:
Server return in payload error code 20001 `{"error_code": 20001}`. This mean some error. I want seen clear message  in exception:
```python
class WrongCardNumber(APIRequestError):
    message = "Wrong card number"

ERROR_CODES_WITH_EXCEPTIONS_MAP = {
    20001: WrongCardNumber
}

class MyExceptionHandler(ExceptionHandler):
    @staticmethod
    def get_exception_class(response: Response) -> Type[APIRequestError]:
        return ERROR_CODES_WITH_EXCEPTIONS_MAP.get(response.json().get('error_code'))
```

Reply:
okay, I'm happy with this.